packages/theme-language-server-common/src/css/CSSLanguageService.ts

```diff
@@ -0,0 +1,129 @@
+import { LiquidRawTag, NodeTypes } from '@shopify/liquid-html-parser';
+import { Mode, Modes, SourceCodeType, findCurrentNode } from '@shopify/theme-check-common';
+import { LanguageService, Stylesheet, getCSSLanguageService } from 'vscode-css-languageservice';
+import {
+  CompletionItem,
+  CompletionList,
+  CompletionParams,
+  Diagnostic,
+  DocumentDiagnosticParams,
+  Hover,
+  HoverParams,
+  ClientCapabilities as LSPClientCapabilities,
+} from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { DocumentManager } from '../documents';
+
+export class CSSLanguageService {
+  private services: Record<Mode, LanguageService | null>;
+
+  constructor(
+    private documentManager: DocumentManager,
+    private getModeForURI: (uri: string) => Promise<Mode>,
+  ) {
+    this.services = { [Modes[0]]: null, [Modes[1]]: null };
+  }
+
+  async setup(clientCapabilities: LSPClientCapabilities) {
+    await Promise.all(
+      Modes.map(async (mode) => {
+        const service = getCSSLanguageService({
+          clientCapabilities,
+        });
+
+        this.services[mode] = service;
```

Comment:
It's pretty unclear, ngl. The mode thing is for switching between different JSON schemas between theme app extensions JSON schemas for the `{% schema %}` tag inside block files. I don't think we need a mode switch for theme app extensions? 

The TL;DR is that theme app extensions look a bit like themes
```
assets/
blocks/
snippets/
```

But, unfortunately, the `{% schema %}` JSON code that they accept isn't 100% the same as the ones we use for themes. And since we wanted to have matching of file patterns -> JSON schemas, we needed the mode switch to target `blocks/*.liquid` files with different JSON schemas. 

I don't think we need that for CSS or JS.

Reply:
Removed it